code/modules/tgui/modules/appearance_changer.dm

```diff
@@ -91,6 +91,8 @@
 		if("race")
 			if(can_change(APPEARANCE_RACE) && (params["race"] in valid_species))
 				if(target.change_species(params["race"]))
+					if(params["race"] == "Custom Species")
+						target.custom_species = input(usr,"Custom Species name:","Species Name")
```

Comment:
This needs sanitization. As it is, you can put crazy things in here, like html img tags or anything. Run it through sanitize, and allow users to cancel if they change their mind as well.

Reply:
I added the cancel option, but in this case there's no good way to cancel selecting the race, thus if you do hit cancel it simple defaults the custom race to "Custom Race".

I'm probably going to do a rework of this panel at some point to make it easier to use.